rotkehlchen/tasks/assets.py

```diff
@@ -244,3 +270,67 @@ def update_owned_assets(user_db: DBHandler) -> None:
             (DBCacheStatic.LAST_OWNED_ASSETS_UPDATE.value, str(ts_now())),
         )
+
+
+def update_aave_v3_underlying_assets(chains_aggregator: 'ChainsAggregator') -> None:
+    """Fetch the Aave v3 underlying assets and polulate `underlying_tokens_list` in globaldb"""
+    for chain_id, data_provider_address in (
+        (ChainID.ETHEREUM, AAVE_V3_DATA_PROVIDER_ETH),
+        (ChainID.OPTIMISM, AAVE_V3_DATA_PROVIDER_EVM),
+        (ChainID.POLYGON_POS, AAVE_V3_DATA_PROVIDER_EVM),
+        (ChainID.ARBITRUM_ONE, AAVE_V3_DATA_PROVIDER_EVM),
+        (ChainID.BASE, AAVE_V3_DATA_PROVIDER_BASE),
+        (ChainID.GNOSIS, AAVE_V3_DATA_PROVIDER_GNO),
+        (ChainID.SCROLL, AAVE_V3_DATA_PROVIDER_SCRL),
+    ):
+        node_inquirer = chains_aggregator.get_evm_manager(chain_id=chain_id).node_inquirer  # type: ignore[arg-type]  # all iterated ChainIDs are supported
+        aave_data_provider = node_inquirer.contracts.contract(address=data_provider_address)
+
+        try:
+            underlying_tokens = [  # get all the underlying tokens in aave v3
+                to_checksum_address(token_address)
+                for _, token_address in aave_data_provider.call(
+                    node_inquirer=node_inquirer,
+                    method_name='getAllReservesTokens',
+                )
+            ]
+            reserve_tokens_addresses = node_inquirer.multicall(calls=[
+                (  # get all the reserve tokens for each underlying token
+                    aave_data_provider.address,
+                    aave_data_provider.encode(
+                        method_name='getReserveTokensAddresses',
+                        arguments=[underlying_token],
+                    ),
+                )
+                for underlying_token in underlying_tokens
+            ])
+        except RemoteError as e:
+            log.error(f'Failed to query Aave v3 reserve tokens addresses due to {e!s}')
+            continue
+
+        # add the mappings of underlying tokens to the globaldb
+        for underlying_idx, underlying_token_address in enumerate(underlying_tokens):
+            for decoded_reserve_token_address in aave_data_provider.decode(
+                method_name='getReserveTokensAddresses',
+                arguments=[underlying_token_address],
+                result=reserve_tokens_addresses[underlying_idx],
+            ):
+                try:
+                    decoded_reserve_token = get_or_create_evm_token(
+                        evm_inquirer=node_inquirer,
+                        userdb=node_inquirer.database,
+                        evm_address=to_checksum_address(decoded_reserve_token_address),
+                        chain_id=chain_id,
+                        protocol=CPT_AAVE_V3,
+                        token_kind=EvmTokenKind.ERC20,
+                        underlying_tokens=[UnderlyingToken(
+                            address=underlying_token_address,
```

Comment:
don't you need to ensure that the underlying token exists? Otherwise the price won't be queried correctly. Also I'm not sure if the relation is a foreignkey... I would assume so

Reply:
It's ensured in the handler function here [# make sure underlying token address is tracked if not already there](https://github.com/OjusWiZard/rotki/blob/fb0d30f60d3b33807dd974cae127ab7a1dff9c2e/rotkehlchen/globaldb/handler.py#L737)
And, referenced using the foreign key, [FOREIGN KEY(identifier) REFERENCES evm_tokens(identifier) ON UPDATE CASCADE ON DELETE CASCADE](https://github.com/OjusWiZard/rotki/blob/fb0d30f60d3b33807dd974cae127ab7a1dff9c2e/rotkehlchen/globaldb/schema.py#L8)

Reply:
add a comment for this then next to here. I had the same question.